src/TimerContext.tsx

```diff
@@ -0,0 +1,211 @@
+                                } else {
+                                    timer.isWorking = true;
+                                    timer.duration = timer.workTime;
+                                }
+                            }
+                        }
+                        break;
+                }
+
+                return updatedTimers;
+            });
+        }, 10);
+
+        return () => clearInterval(intervalId);
+    }, [currentTimerIndex, timers]);
+
+    const addTimer = (timer: Timer) => {
+        setTimers(prev => [...prev, timer]);
+    };
+
+    const removeTimer = (id: string) => {
+        setTimers(prev => prev.filter(timer => timer.id !== id));
+        if (currentTimerIndex !== null) {
+            setCurrentTimerIndex(prev => (prev !== null && timers[prev].id === id ? null : prev));
+        }
+    };
+
+    const resetTimers = () => {
+        setTimers(prev =>
+            prev.map(timer => {
+                switch (timer.type) {
+                    case 'stopwatch':
+                        return { ...timer, duration: 0, status: 'not running' };
+                    case 'countdown':
+                        return { ...timer, duration: timer.initialDuration, status: 'not running' };
+                    case 'XY':
+                    case 'tabata':
+                        return {
+                            ...timer,
+                            duration: timer.workTime,
+                            currentRound: 1,
+                            isWorking: true,
+                            status: 'not running',
+                        };
+                }
+            }),
+        );
+        setCurrentTimerIndex(null);
+    };
+
+    const toggleStartPause = () => {
+        if (currentTimerIndex === null && timers.length > 0) {
+            setCurrentTimerIndex(0);
+            setTimers(prev => prev.map((timer, index) => (index === 0 ? { ...timer, status: 'running' } : timer)));
+        } else if (currentTimerIndex !== null) {
+            setTimers(prev => prev.map((timer, index) => (index === currentTimerIndex ? { ...timer, status: timer.status === 'running' ? 'paused' : 'running' } : timer)));
+        }
+    };
+
+    const fastForward = () => {
+        if (currentTimerIndex === null) return;
+
+        setTimers(prev => prev.map((timer, index) => (index === currentTimerIndex ? { ...timer, status: 'completed' } : timer)));
+
+        const nextIndex = currentTimerIndex + 1;
+        if (nextIndex < timers.length) {
+            setCurrentTimerIndex(nextIndex);
+            setTimers(prev => prev.map((timer, index) => (index === nextIndex ? { ...timer, status: 'running' } : timer)));
+        } else {
+            setCurrentTimerIndex(null);
+        }
+    };
+
+    const getTotalTime = () => {
+        return timers.reduce((total, timer) => {
+            switch (timer.type) {
+                case 'stopwatch':
+                    return total + timer.duration;
```

Comment:
I'm confused because the stopwatch UI doesn't allow entering a duration

Reply:
You are right. Implemented. I guess it makes sense since it is a stopwatch, and it is not predetermined.